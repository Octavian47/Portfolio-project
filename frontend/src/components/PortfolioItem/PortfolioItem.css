/* Container for the entire portfolio item */
/* Portfolio Item Container */
.portfolio-item {
  background-color: #fff;             /* White background for the item */
  border: 1px solid #ddd;             /* Light gray border */
  border-radius: 8px;                   /* Rounded corners */
  padding: 16px;                        /* Padding inside the item */
  margin: 0 auto;                       /* Center the item in the grid cell */
  text-align: center;                    /* Center-align text */
  box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1); /* Subtle shadow for depth */
  transition: transform 0.3s ease, box-shadow 0.3s ease; /* Smooth hover transition */
  max-width: 100%;                       /* Ensure the item doesn't exceed its container */
}

/* Portfolio Item Image */
.portfolio-image {
  width: 100%;                        /* Image takes full width of the container */
  max-width: 600px;                   /* Set a max-width for the image */
  height: auto;                       /* Maintain aspect ratio */
  border-radius: 8px;                 /* Rounded corners for the image */
  margin-bottom: 12px;                /* Space below the image */
}
  
  /* Title styling */
  .portfolio-item h3 {
    font-size: 1.5em;                  /* Larger font for title */
    margin-bottom: 8px;                /* Space below the title */
    color: #333;                     /* Dark gray color for the title */
  }
  
  /* Description styling */
  .portfolio-item p {
    font-size: 1em;                    /* Standard font size for description */
    color: #666;                     /* Medium gray color for description text */
    margin-bottom: 16px;               /* Space below the description */
  }
  
  /* Link to the client's site */
  .portfolio-item a {
    font-size: 1em;                    /* Standard font size for link */
    color: #007BFF;                  /* Blue color for link */
    text-decoration: none;             /* Remove underline from link */
    display: inline-block;             /* Inline-block for hover effects */
    margin-top: 8px;                   /* Space above the link */
    transition: color 0.3s ease;       /* Smooth color transition on hover */
  }
  
  /* Hover effects for the link */
  .portfolio-item a:hover {
    color: #0056b3;                    /* Darker blue on hover */
  }
  
  /* Hover effects for the entire portfolio item */
  .portfolio-item:hover {
    transform: translateY(-5px);       /* Slight lift effect on hover */
    box-shadow: 0 6px 12px rgba(0, 0, 0, 0.1); /* Deepen shadow on hover */
  }
  
  /* Responsive design: Adjust grid layout for smaller screens */
  @media (max-width: 600px) {
    .portfolio-item {
      margin: 8px;                     /* Reduce margin on small screens */
      padding: 12px;                   /* Reduce padding on small screens */
    }
  
    .portfolio-item h3 {
      font-size: 1.2em;                /* Slightly smaller font for title */
    }
  
    .portfolio-item p,
    .portfolio-item a {
      font-size: 0.9em;                /* Slightly smaller font for text and links */
    }
  }
  