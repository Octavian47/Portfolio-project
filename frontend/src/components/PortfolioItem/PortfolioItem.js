import React from 'react';
import './PortfolioItem.css';

const PortfolioItem = ({ title, description, imageFilename, clientUrl, status }) => {
  // Do not render the component if the status is 'hidden'
  if (status === 'hidden') {
    return null;
  }

  return (
    <div className="portfolio-item">
      <img src={imageFilename} alt={title} className="portfolio-image" />
      <h3>{title}</h3>
      <p>{description}</p>
      <a href={clientUrl} target="_blank" rel="noopener noreferrer">
        Visit Client's Site
      </a>
    </div>
  );
};

export default PortfolioItem;
