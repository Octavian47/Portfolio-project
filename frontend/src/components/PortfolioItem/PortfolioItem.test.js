import React from 'react';
import { render } from '@testing-library/react';
import PortfolioItem from './PortfolioItem';

describe('PortfolioItem Component', () => {

  it('renders the PortfolioItem with the correct title, description, and link', () => {
    const { getByText } = render(
      <PortfolioItem 
        title="Test Title"
        description="Test Description"
        imageFilename="/path/to/image.jpg"
        clientUrl="https://www.example.com"
        status="visible"
      />
    );

    // Check if the title, description, and link text are rendered correctly
    expect(getByText('Test Title')).toBeInTheDocument();
    expect(getByText('Test Description')).toBeInTheDocument();
    expect(getByText("Visit Client's Site")).toBeInTheDocument();
  });

  it('renders the image with the correct src and alt attributes', () => {
    const { getByAltText } = render(
      <PortfolioItem 
        title="Test Title"
        description="Test Description"
        imageFilename="/path/to/image.jpg"
        clientUrl="https://www.example.com"
        status="visible"
      />
    );

    // Check if the image has the correct src and alt attributes
    const image = getByAltText('Test Title');
    expect(image).toHaveAttribute('src', '/path/to/image.jpg');
    expect(image).toHaveAttribute('alt', 'Test Title');
  });

  it('renders the link with the correct href attribute', () => {
    const { getByText } = render(
      <PortfolioItem 
        title="Test Title"
        description="Test Description"
        imageFilename="/path/to/image.jpg"
        clientUrl="https://www.example.com"
        status="visible"
      />
    );

    // Check if the link has the correct href attribute
    const link = getByText("Visit Client's Site").closest('a');
    expect(link).toHaveAttribute('href', 'https://www.example.com');
    expect(link).toHaveAttribute('target', '_blank');
    expect(link).toHaveAttribute('rel', 'noopener noreferrer');
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(
      <PortfolioItem 
        title="Test Title"
        description="Test Description"
        imageFilename="/path/to/image.jpg"
        clientUrl="https://www.example.com"
        status="visible"
      />
    );

    // Match the component against the snapshot
    expect(asFragment()).toMatchSnapshot();
  });

});
