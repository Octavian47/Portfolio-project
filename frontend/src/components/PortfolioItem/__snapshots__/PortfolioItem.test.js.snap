// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortfolioItem Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="portfolio-item"
  >
    <img
      alt="Test Title"
      class="portfolio-image"
      src="/path/to/image.jpg"
    />
    <h3>
      Test Title
    </h3>
    <p>
      Test Description
    </p>
    <a
      href="https://www.example.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      Visit Client's Site
    </a>
  </div>
</DocumentFragment>
`;
